sklearn/utils/estimator_checks.py

```diff
@@ -72,6 +72,36 @@ def _safe_tags(estimator, key=None):
     return _DEFAULT_TAGS
 
 
+def _raise_xfail(reason, request=None):
+    """Mark a check as a known failure
+
+    This function is mostly useful to be able to skip the test
+    if pytest is not installed. Outside of a scikit-learn pytest
+    session the SkipTest exception will allways be raised.
+
+    Parameters
+    ----------
+    reason : str
+        reason for the known failure
+    request: default=None
+        result of the pytest request fixture.
+    """
+    if not getattr(sys, "_is_pytest_session", False):
+        # outside of a scikit-learn pytest session always raise SkipTest
+        raise SkipTest('XFAIL ' + str(reason))
+    try:
+        import pytest
+        if request is None:
+            # raise XFAIL and halt execution
+            pytest.xfail(reason)
+        else:
+            # mark test as XFAIL and continue excecution to see if it will
+            # actually fail.
+            request.applymarker(pytest.mark.xfail(reason=reason))
+    except ImportError:
```

Comment:
time to make pytest a dependency for the whole test suite ? :D 
(I'm kidding I don't want to start a discussion here)